jerry-core/jmem/jmem-heap.c

```diff
@@ -411,11 +406,61 @@ jmem_heap_alloc_block (const size_t size)
   }
 
   JERRY_ASSERT (data_space_p == NULL);
+  return data_space_p;
+} /* jmem_heap_gc_and_alloc_block */
+
+
+/**
+ * Allocation of memory block, running 'try to give memory back' callbacks, if there is not enough memory.
+ *
+ * Note:
+ *      If there is still not enough memory after running the callbacks, then the engine will be
+ *      terminated with ERR_OUT_OF_MEMORY.
+ *
+ * @return NULL, if the required memory is 0
+ *         pointer to allocated memory block, otherwise
+ */
+void * __attribute__((hot))
+jmem_heap_alloc_block (const size_t size)  /**< required memory size */
```

Comment:
I would implement this a bit differently. I would add a bool parameter to jmem_heap_gc_and_alloc_block to tell whether it should return with NULL or error, and make these functions a one-liner always inline functionsa, and just set that bool parameter.